test/extensions/filters/listener/original_dst/original_dst_integration_test.cc

```diff
@@ -0,0 +1,144 @@
+#include "envoy/config/bootstrap/v3/bootstrap.pb.h"
+#include "envoy/event/dispatcher.h"
+#include "envoy/extensions/filters/listener/original_dst/v3/original_dst.pb.h"
+#include "envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.pb.h"
+#include "envoy/http/header_map.h"
+
+#include "source/common/http/headers.h"
+#include "source/common/network/utility.h"
+
+#include "test/common/upstream/utility.h"
+#include "test/integration/http_integration.h"
+#include "test/integration/http_protocol_integration.h"
+#include "test/integration/utility.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+
+using OriginalDstIntegrationTest = HttpProtocolIntegrationTest;
+
+ConfigHelper::ConfigModifierFunction setOriginalDstCluster(int port) {
```

Comment:
Just a high-level feedback: I think the tricky part about OriginalDST is the host management. That means creation of host on-demand, deletion of the host after idle/cleanup, and re-creation of the host. With the port override, it's just one host, so we need to make sure pools remove them in-between requests/connections. For example, for TCP, I think we have to wait for "cleanup" interval .